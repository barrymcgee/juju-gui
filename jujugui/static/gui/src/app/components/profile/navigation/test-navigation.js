/* Copyright (C) 2017 Canonical Ltd. */
'use strict';

const React = require('react');

const ProfileNavigation = require('./navigation');

const jsTestUtils = require('../../../utils/component-test-utils');

describe('Profile Navigation', function() {
  const sectionsMap = new Map([
    ['models', {label: 'Models'}],
    ['charms', {label: 'Charms'}]
  ]);

  function renderComponent(options) {
    return jsTestUtils.shallowRender(
      <ProfileNavigation
        activeSection="bundles"
        changeState={options.changeState || sinon.stub()}
        sectionsMap={sectionsMap} />, true);
  }

  it('can render', () => {
    const changeState = sinon.stub();
    const output = renderComponent({
      changeState
    }).getRenderOutput();
    const expected = (
      <div className="profile-navigation">
        <ul>
          <li className="profile-navigation__list-item"
            key='models' onClick={sinon.stub()} role="button">
            Models
          </li>
          <li className="profile-navigation__list-item"
            key='charms' onClick={sinon.stub()} role="button">
            Charms
          </li>
        </ul>
      </div>
    );
    expect(output).toEqualJSX(expected);
  });

  it('calls changeState when nav item clicked', () => {
    const changeState = sinon.stub();
    const output = renderComponent({
      changeState
    }).getRenderOutput();
    output.props.children.props.children[1].props.onClick();
    assert.equal(changeState.callCount, 1);
    assert.deepEqual(changeState.args[0][0], {hash: 'charms'});
  });

  it('updates the active nav item when re-rendered', () => {
    const changeState = sinon.stub();
    const renderer = renderComponent({
      changeState
    });
    const output = renderer.getRenderOutput();
    const expected = (
      <div className="profile-navigation">
        <ul>
          <li className="profile-navigation__list-item"
            key='models' onClick={sinon.stub()} role="button">
            Models
          </li>
          <li className="profile-navigation__list-item"
            key='charms' onClick={sinon.stub()} role="button">
            Charms
          </li>
        </ul>
      </div>
    );
    expect(output).toEqualJSX(expected);
    renderer.render(
      <ProfileNavigation
        activeSection="charms"
        changeState={changeState}
        sectionsMap={sectionsMap} />);
    const output2 = renderer.getRenderOutput();
    const expected2 = (
      <div className="profile-navigation">
        <ul>
          <li className="profile-navigation__list-item"
            key='models' onClick={sinon.stub()} role="button">
            Models
          </li>
          <li className="profile-navigation__list-item is-active"
            key='charms' onClick={sinon.stub()} role="button">
            Charms
          </li>
        </ul>
      </div>
    );
    expect(output2).toEqualJSX(expected2);
  });

});
