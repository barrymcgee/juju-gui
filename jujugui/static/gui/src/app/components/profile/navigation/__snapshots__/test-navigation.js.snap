// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Navigation can render 1`] = `
<div
  className="profile-navigation"
>
  <ul
    className="p-list"
  >
    <li
      className="p-list__item models"
      key="models"
      onClick={[Function]}
      role="button"
    >
      Models
    </li>
    <li
      className="p-list__item charms"
      key="charms"
      onClick={[Function]}
      role="button"
    >
      Charms
    </li>
  </ul>
</div>
`;
